rir/src/runtime/FunctionSignature.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include "R/r.h"
+
+namespace rir {
+
+struct FunctionSignature {
+
+    struct ArgumentType {
+        bool isEvaluated = false;
+        unsigned char type = PROMSXP;
+        int length = -1;
+
+        ArgumentType() = default;
+        ArgumentType(SEXP s) : type(TYPEOF(s)) {
+            if (type == PROMSXP) {
+                isEvaluated = PRVALUE(s) != R_UnboundValue;
+            }
+            if (isEvaluated)
+                length = XLENGTH(s);
+        }
+        ArgumentType(bool evaled, unsigned int type) :
```

Comment:
Tbh I don't know what to make of the `lenght`... It is initialized to `-1` via the in-class initializer, and the idea was that this would mean don't know / don't care, the constructor only updating the value if it knows better.

But I agree that it should be commented at least.